src/main/scala/eventstore/PersistentSubscriptionActor.scala

```diff
@@ -0,0 +1,133 @@
+package eventstore
+import akka.actor.Status.Failure
+import akka.actor.{ ActorRef, FSM, Props, Terminated }
+import eventstore.PersistentSubscription.Ack
+import eventstore.{ PersistentSubscription => PS }
+
+object PersistentSubscriptionState {
+
+  sealed trait State
+
+  case object Unsubscribed extends State
+
+  case object LiveProcessing extends State
+
+  case object CatchingUp extends State
+
+}
+
+sealed trait Data
```

Comment:
I'd prefere to have these hidden in companion object and not available to client users